Networking/Networking/Model/Product/ProductVariation.swift

```diff
@@ -0,0 +1,357 @@
+import Foundation
+
+/// Represents a Product Variation Entity.
+///
+public struct ProductVariation: Decodable {
+    public let siteID: Int64
+    public let productID: Int64
+
+    public let productVariationID: Int64
+
+    public let attributes: [ProductVariationAttribute]
+    public let image: ProductImage?
+
+    public let permalink: String
+
+    public let dateCreated: Date
+    public let dateModified: Date?
+    public let dateOnSaleStart: Date?
+    public let dateOnSaleEnd: Date?
+
+    public let statusKey: String        // draft, pending, private, published
```

Comment:
great question! there used to be a calculated property below:
```
public var status: ProductStatus {
        return ProductStatus(rawValue: statusKey)
}
```
and I just removed those computed enum properties to the properties themselves in https://github.com/woocommerce/woocommerce-ios/pull/1492/commits/4166f045ce0eb64e9703efcc90fa86218b5c2b1b

I originally followed what `Product` did, and I think the status key string from the server is kept so that this key string can be stored in the Storage model. However, we can just store the enum's `rawValue` string, so I don't think it's necessary to keep 2 versions of the same field anymore.